java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/api/v1/menu/impl/MenuControllerImpl.java

```diff
@@ -31,18 +41,61 @@ public MenuControllerImpl(MenuService menuService) {
                                                        final Integer pageSize,
                                                        final Integer pageNumber) {
 
+        List<SearchMenuResultItem> menuResultItems;
```

Comment:
Too much logic in the controller. Move it to the service.
The controller should just validate the inputs